test_conformance/extensions/cl_khr_command_buffer/command_buffer_event_sync.cpp

```diff
@@ -603,10 +603,10 @@ struct CommandBufferEventSync : public BasicCommandBufferTest
                                     event_ptrs[1], nullptr);
         test_error(error, "clEnqueueReadBuffer failed");
 
-        error = clFinish(queue);
+        error = clFinish(queue_sec);
         test_error(error, "clFinish failed");
 
-        error = clFinish(queue_sec);
+        error = clFinish(queue);
         test_error(error, "clFinish failed");
```

Comment:
Looking at this again, could we actually just set blocking to `CL_TRUE`  on the `clEnqueueReadBuffer` call above and then remove these two `clFinish` calls

Reply:
In that case we would have the same problem again, setting `clEnqueueReadBuffer` to be a blocking command is the same as having clFinish(queue).

Reply:
ok cool, I feel like I'd expect these sorts of synchronization operations to transitively wait on the dependencies in other queues to complete, but that's behaviour from the core spec, and if it's ambiguous we shouldn't start mandating it in order to pass testing with this extension.